docs/publishing/index.md

```diff
@@ -0,0 +1,123 @@
+
+Keep in mind that these suggestions are not specifically tuned for publishing to a CDN or other contexts where broader application-specific optimizations will not be applied to the code available therein.
+
+<style>
+.content__default ul {list-style:none;padding:0;}
+</style>
+
+### Do: 👍
+
+- ✅ [Do publish latest standard EcmaScript](#do-publish-latest-standard-ecmascript)
+- ✅ [Do publish standard es modules](#do-publish-standard-es-modules)
+- ✅ [Do include appropriate `package.json` entries](#do-include-appropriate-packagejson-entries)
+- ✅ [Do export element classes](#do-export-element-classes)
+- ✅ [Do export side effects separately](#do-export-side-effects-separately)
+- ✅ [Do keep side effects out of your index.js](#do-keep-side-effects-out-of-your-indexjs)
+- ✅ [Do import 3rd party node modules with "bare" import specifiers](#do-import-3rd-party-node-modules-with-bare-import-specifiers)
+- ✅ [Do include file extensions in import specifiers, always](#do-include-file-extensions-in-import-specifiers-always)
+- _Optional_: [use `type: "module"` when possible](#optional-use-type-module)
+- _Optional_: [include typings; e.g. `*.d.ts` files](#optional-include-typings)
+- _Optional_: [include source maps](#optional-include-source-maps)
+
+### Don't: 👎
+
+- ❌ [Do not bundle](#do-not-bundle)
+- ❌ [Do not minify](#do-not-minify)
+- ❌ [Do not optimize](#do-not-optimize)
+- ❌ [Do not use `.mjs` file extensions](#do-not-use-mjs-file-extensions)
+- ❌ [Do not import polyfills](#do-not-import-polyfills)
+
+## Do publish latest standard EcmaScript
+
+Contrary to some long-standing beliefs, an implementing project will always know more about the delivery context of the code it leverages than the developer of any of its dependencies. Taking this fact into account, the act of transpiling your code below a reasonably modern level (e.g. [TC39 Stage 4](https://github.com/tc39/proposals/blob/master/finished-proposals.md) and/or available cross-browser) is the best way that you can prepare your package to empower downstream projects to make their own decisions about the code they ship to their users.
```

Comment:
While I'm a big fan of your style of writing, I feel like we need to be extremely specific and clear in these instructions, and we may overall need to "dumb it down" somewhat. 

In this paragraph specifically this sentence could be simplified a lot:
> "an implementing project will always know more about the delivery context of the code it leverages than the developer of any of its dependencies."